Framework/DataHandling/src/SaveOpenGenieAscii.cpp

```diff
@@ -49,337 +46,382 @@ void SaveOpenGenieAscii::init() {
+  const std::string yunitsVal = "Neutron counts / \\\\gms";
+  addToOutputBuffer("yunits", m_stringType, yunitsVal);
+  addToOutputBuffer("ylabel", m_stringType, yunitsVal);
+}
 
-    if (type.std::string::find("vector") &&
-        type.std::string::find("double") != std::string::npos) {
+/**
+  * Calculates the delta in the logged (i.e. not data) X Y Z values by
+  * from the minimum and maximum logged values
+  */
+void SaveOpenGenieAscii::calculateXYZDelta(const std::string &unit,
+                                           const Kernel::Property *values) {
+  // Cast to TimeSeries so we can use the min/max value methods
+  const auto positionValues =
+      static_cast<const TimeSeriesProperty<double> *>(values);
+
+  const double deltaValue =
+      positionValues->maxValue() - positionValues->minValue();
+
+  addToOutputBuffer('d' + unit, m_floatType, std::to_string(deltaValue));
+}
 
-      auto tsp = ws->run().getTimeSeriesProperty<double>(name);
-      value = boost::lexical_cast<std::string>(tsp->timeAverageValue());
+/**
+* Converts the workspace data into strings which are compatible
+* with OpenGenie and stores them into a tuple with the number
+* of data points for X Y and E
+*
+* @return :: Returns a vector of tuples - the tuples being the raw
+* data as a string and the number of data points. The first entry in
+* the vector is X, second Y, and final element E data
+*
+*/
+std::vector<std::tuple<std::string, int>>
+SaveOpenGenieAscii::convertWorkspaceToStrings() {
+  // Padding to apply after 10 data values
+  const std::string newlineStr = "\r\n    ";
+
+  // Build x, y and e strings - first 4 spaces for correct padding
+  std::string xValsOutput("    "), yValsOutput("    "), eValsOutput("    ");
+  // Number of values seen also used to track 10 values for a newline
+  int xCount(0), yCount(0), eCount(0);
+
+  const auto &x = m_inputWS->x(0);
+  for (const auto xVal : x) {
+    if (xCount % 10 == 0) {
+      xValsOutput += newlineStr;
     }
+    xCount++;
+    xValsOutput += std::to_string(xVal) + ' ';
+  }
 
-    else if (type.std::string::find("vector") &&
-             type.std::string::find("int") != std::string::npos) {
-
-      auto tsp = ws->run().getTimeSeriesProperty<int>(name);
-      value = boost::lexical_cast<std::string>(tsp->timeAverageValue());
+  const auto &y = m_inputWS->y(0);
+  for (const auto yVal : y) {
+    if (yCount % 10 == 0) {
+      yValsOutput += newlineStr;
     }
+    yCount++;
+    yValsOutput += std::to_string(yVal) + ' ';
+  }
 
-    else if (type.std::string::find("vector") &&
-             type.std::string::find("bool") != std::string::npos) {
-
-      auto tsp = ws->run().getTimeSeriesProperty<bool>(name);
-      value = boost::lexical_cast<std::string>(tsp->timeAverageValue());
+  const auto &e = m_inputWS->e(0);
+  for (const auto eVal : e) {
+    if (eCount % 10 == 0) {
+      eValsOutput += newlineStr;
     }
+    eCount++;
+    eValsOutput += std::to_string(eVal) + ' ';
+  }
```

Comment:
I'm not too sure how to go about this - originally I wrote a small method which took the data as a parameter and was just called like so:
```c++
const auto &x = m_inputWS-x(0)
processWorkspaceData(x)
```
However I can't seem to see a universal histogram type for the signature - I have to either specify `HistogramX/Y/E`?